adminSiteServer/apiRouter.ts

```diff
@@ -1765,34 +1765,60 @@ apiRouter.delete("/variables/:variableId", async (req: Request) => {
 apiRouter.get("/datasets.json", async (req) => {
     const datasets = await db.queryMysql(`
         SELECT
-            d.id,
-            d.namespace,
-            d.name,
-            d.description,
-            d.dataEditedAt,
+            ad.id,
+            ad.namespace,
+            ad.name,
+            d.shortName,
+            ad.description,
+            ad.dataEditedAt,
             du.fullName AS dataEditedByUserName,
-            d.metadataEditedAt,
+            ad.metadataEditedAt,
             mu.fullName AS metadataEditedByUserName,
-            d.isPrivate,
-            d.nonRedistributable
-        FROM active_datasets d
-        JOIN users du ON du.id=d.dataEditedByUserId
-        JOIN users mu ON mu.id=d.metadataEditedByUserId
-        ORDER BY d.dataEditedAt DESC
+            ad.isPrivate,
+            ad.nonRedistributable,
+            d.version
+        FROM active_datasets ad
+        JOIN users du ON du.id=ad.dataEditedByUserId
+        JOIN users mu ON mu.id=ad.metadataEditedByUserId
+        JOIN datasets d ON d.id=ad.id
+        ORDER BY ad.dataEditedAt DESC
```

Comment:
Instead of getting the counts separately below and having to merge them locally it's more efficient and IMHO nicer to let SQL do this:
(suggestions don't work here for some reason)
```sql
with variable_counts as
(
SELECT
	v.datasetId as datasetId,
	COUNT(DISTINCT cd.chartId) as numCharts
FROM
	chart_dimensions cd
JOIN variables v ON
	cd.variableId = v.id
GROUP BY
	v.datasetId)
SELECT
	ad.id,
	ad.namespace,
	ad.name,
	d.shortName,
	ad.description,
	ad.dataEditedAt,
	du.fullName AS dataEditedByUserName,
	ad.metadataEditedAt,
	mu.fullName AS metadataEditedByUserName,
	ad.isPrivate,
	ad.nonRedistributable,
	d.version,
	vc.numCharts
FROM
	active_datasets ad
join variable_counts vc on
	ad.id = vc.datasetId
JOIN users du ON
	du.id = ad.dataEditedByUserId
JOIN users mu ON
	mu.id = ad.metadataEditedByUserId
JOIN datasets d ON
	d.id = ad.id
ORDER BY
	ad.dataEditedAt DESC
```

This uses a CTE (Common Table Expression) to basically define a temporary table that we then join with to get the count for each active dataset. The temporary table with the chart counts per dataset is constructed from the chart_dimensions table which is in a way a materialized "cache" of charts.config '$->>dimensions'.

Reply:
I didn't know about chart_dimensions, that's brilliant!
Thank you for taking the time to write this down 🙏🏻  (I'm very keen to learn more sql stuff)